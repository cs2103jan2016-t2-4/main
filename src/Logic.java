
import java.util.LinkedList;
import java.util.List;

public class Logic {

	private static final String FILE_NAME = "WURI.txt";

	private static final String MESSAGE_TASK_ADDED = "added to %1$s: \"%2$s\"";
	private static final String MESSAGE_TASK_EDITED = "Edited task %1$s";
	private static final String MESSAGE_TASK_DELETED = "deleted from %1$s: \"%2$s\"\n";
	private static final String MESSAGE_SEARCH_NO_RESULT = "Did not find any phrase with the keywords\n";

	private static final String MESSAGE_INVALID_INDEX = "Invalid index\n";
	private static final String MESSAGE_INVALID_ARGUMENTS = "Invalid arguments %1$s\n";
	
	
	

	private List<Task> _taskList = new LinkedList<Task>();

	public CommandDetails executeCommand(String userInput) {
		Parser parser = new Parser(userInput);
		CommandDetails commandDetails = new CommandDetails(parser.getCommandTypeStr());
		switch (commandDetails.getCommandType()) {
			case ADD :
				addTask(parser, commandDetails);
				break;

			case EDIT :
				editTask(parser, commandDetails);
				break;

			case MARK_AS_COMPLETE :
			//	markTaskAsComplete(parser, commandDetails);
				break;

			case DELETE :
				// todo
				break;

			case FIND :
				// todo
				break;

			case UNDO :
				// todo
				break;

			case QUIT :
				break;
		}
		return commandDetails;
	}

	private void addTask(Parser parser, CommandDetails commandDetails) {
		Task newTask = new Task(parser.getTaskDescription());
		newTask.setStartDate(parser.getStartDate());
		newTask.setEndDate(parser.getEndDate());

		_taskList.add(newTask);

		String taskStr = newTask.toString();
		commandDetails.setTaskStr(taskStr);
		commandDetails.setFeedback(String.format(MESSAGE_TASK_ADDED, taskStr));
	}
	
	private void editTask(Parser parser, CommandDetails commandDetails) {
		int taskIndex = parser.getTaskIndex();
		
		Task task = _taskList.get(taskIndex);
	
		task.setStartDate(parser.getStartDate());
		task.setEndDate(parser.getEndDate());
			
		task.setDescription(parser.getTaskDescription());
		
		commandDetails.setTaskIndex(taskIndex);
		commandDetails.setTaskStr(task.toString());
		commandDetails.setFeedback(String.format(MESSAGE_TASK_EDITED, taskIndex));
	}
	
	
}
