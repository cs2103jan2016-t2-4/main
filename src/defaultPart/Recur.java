package defaultPart;

public class Recur {

	public enum TimeUnit {
		DAY, WEEK, MONTH, YEAR
	}

	private TimeUnit _timeUnit;
	private int _frequency = 1;
	private TaskDate _endDate;
	private TaskDate _startDate;

	public Recur() {

	}

	public TimeUnit getTimeUnit() {
		return _timeUnit;
	}

	public void setTimeUnit(TimeUnit timeUnit) {
		_timeUnit = timeUnit;
	}

	public int getFrequency() {
		return _frequency;
	}

	public void setFrequency(int frequency) {
		_frequency = frequency;
	}

	public TaskDate getEndDate() {
		return _endDate;
	}

	public void setEndDate(TaskDate endDate) {
		_endDate = endDate;
	}

	public boolean willRecur() {
		return (this.getNextRecur() != null);
	}

	public TaskDate getNextRecur() {
		TaskDate nextDate = (TaskDate) getStartDate().clone();
		/*
		 * TaskDate today = initializeToday(); getNextRecurAfterToday(nextDate, today);
		 */

		incrementNextDate(nextDate);

		if (nextDateAfterEndDate(nextDate)) {
			return null;
		}
		return nextDate;
	}

	private void incrementNextDate(TaskDate nextDate) {
		switch (_timeUnit) {
			case DAY :
				nextDate.add(TaskDate.DATE, this._frequency);
				break;
			case WEEK :
				nextDate.add(TaskDate.DATE, this._frequency * 7);
				break;
			case MONTH :
				nextDate.add(TaskDate.MONTH, this._frequency);
				break;
			case YEAR :
				nextDate.add(TaskDate.YEAR, this._frequency);
				break;
		}
		nextDate.getTimeInMillis();
	}

	private boolean nextDateAfterEndDate(TaskDate nextDate) {
		return _endDate != null && nextDate.compareTo(_endDate) > 0;
	}

	private void getNextRecurAfterToday(TaskDate nextDate, TaskDate today) {
		while (nextDateBeforeToday(nextDate, today)) {
			incrementNextDate(nextDate);
		}
	}

	private boolean nextDateBeforeToday(TaskDate nextDate, TaskDate today) {
		return nextDate.compareTo(today) < 0;
	}

	private TaskDate initializeToday() {
		// initialize "today" to 00:00am of tomorrow
		TaskDate today = new TaskDate();
		today.set(TaskDate.DATE, today.get(TaskDate.DATE) + 1);
		today.set(TaskDate.HOUR_OF_DAY, 0);
		today.set(TaskDate.MINUTE, 0);
		today.set(TaskDate.SECOND, 0);
		today.set(TaskDate.MILLISECOND, 0);
		today.getTimeInMillis();
		return today;
	}

	@Override
	public String toString() {
		String formattedDate = _endDate == null ? "" : _endDate.toString();
		return _frequency + "" + _timeUnit.name().toLowerCase().charAt(0) + " " + formattedDate;
	}

	public TaskDate getStartDate() {
		return _startDate;
	}

	public void setStartDate(TaskDate startDate) {
		_startDate = startDate;
	}
}
