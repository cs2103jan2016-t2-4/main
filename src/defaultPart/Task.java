package defaultPart;

import java.text.SimpleDateFormat;
import java.util.Calendar;

public class Task {

	private String _description;

	private Calendar _date;
	private Calendar _startTime;
	private Calendar _endTime;

	private Recur _recur;
	private boolean _isCompleted;

	public void setDescription(String description) {
		_description = description;
	}

	public Calendar getDate() {
		return _date;
	}

	public void setDate(Calendar date) {
		_date = date;
	}

	public Calendar getStartTime() {
		return _startTime;
	}

	public void setStartTime(Calendar startTime) {
		_startTime = startTime;
	}

	public Calendar getEndTime() {
		return _endTime;
	}

	public void setEndTime(Calendar endTime) {
		_endTime = endTime;
	}

	public String getDescription() {
		return _description;
	}

	public Recur getRecur() {
		return _recur;
	}

	public void setRecur(Recur recur) {
		_recur = recur;
	}

	public boolean isCompleted() {
		return _isCompleted;
	}

	public void toggleCompleted() {
		_isCompleted = !_isCompleted;
	}

	@Override
	public String toString() {
		SimpleDateFormat dateFormat = new SimpleDateFormat("dd-MM-yyyy");
		String formattedDate = "";
		if (_date != null) {
			formattedDate = " | " + dateFormat.format(_date.getTime());
		}
		String timeString = "";
		if (_startTime != null) {
			SimpleDateFormat timeFormat = new SimpleDateFormat("K.mma");
			timeString = " | " + timeFormat.format(_startTime.getTime());
			if (_endTime != null) {
				timeString += "-" + timeFormat.format(_endTime.getTime());
			}
		}
		String recurString = "";
		if (_recur != null) {
			recurString = " | recur=" + _recur;
		}
		return _description + formattedDate + timeString + recurString;
	}
}
