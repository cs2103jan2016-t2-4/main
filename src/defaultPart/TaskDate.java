package defaultPart;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.GregorianCalendar;

public class TaskDate extends GregorianCalendar {
	private static final SimpleDateFormat dateFormat = new SimpleDateFormat("dd-MM-yy");

	@Override
	public boolean equals(Object taskDate) {
		return get(Calendar.DAY_OF_MONTH) == ((Calendar) taskDate).get(Calendar.DAY_OF_MONTH)
				&& get(Calendar.MONTH) == ((Calendar) taskDate).get(Calendar.MONTH)
				&& get(Calendar.YEAR) == ((Calendar) taskDate).get(Calendar.YEAR);
	}
	
	@Override
	public String toString() {
		return dateFormat.format(this.getTime());
	}
}
