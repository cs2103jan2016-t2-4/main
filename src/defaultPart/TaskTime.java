package defaultPart;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.GregorianCalendar;

public class TaskTime extends GregorianCalendar {
	private static final SimpleDateFormat timeFormat = new SimpleDateFormat("hh:mma");

	@Override
	public boolean equals(Object taskTime) {
		return get(Calendar.HOUR_OF_DAY) == ((Calendar) taskTime).get(Calendar.HOUR_OF_DAY)
				&& get(Calendar.MINUTE) == ((Calendar) taskTime).get(Calendar.MINUTE);
	}

	@Override
	public String toString() {
		return timeFormat.format(this.getTime());
	}
	
	public void setTimeFromString(String timeString) throws ParseException {
		timeFormat.parse(timeString);
	}
}