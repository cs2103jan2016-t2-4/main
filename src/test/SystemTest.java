package test;

import java.io.File;
import java.io.FileReader;
import java.io.IOException;
import java.text.ParseException;

import org.custommonkey.xmlunit.XMLAssert;
import org.custommonkey.xmlunit.XMLUnit;
import org.junit.Test;
import org.xml.sax.SAXException;

import defaultPart.Logic;
import defaultPart.Recur;
import defaultPart.Recur.TimeUnit;
import defaultPart.Storage;
import defaultPart.Task;
import defaultPart.TaskDate;
import defaultPart.TaskTime;

public class SystemTest {

	/* Location to load/save the expected test results */
	private static final String EXPECTED_FILE_NAME = "test\\SystemTest_expected.xml";
	private static final String TEST_FILE_NAME = "test\\SystemTest_actual.xml";

	/**
	 * Helper function to create the a recur item based on presence of parameters
	 * 
	 * @throws ParseException
	 */
	private Recur createRecur(String timeUnit, int frequency, String startOfRecur, String endOfRecur)
			throws ParseException {
		Recur newRecur = new Recur();

		newRecur.setTimeUnit(TimeUnit.valueOf(timeUnit));
		newRecur.setFrequency(frequency);
		TaskDate startDate = new TaskDate();
		startDate.setDateFromString(startOfRecur);
		newRecur.setStartDate(startDate);
		if (endOfRecur != null) {
			TaskDate endDate = new TaskDate();
			endDate.setDateFromString(endOfRecur);
			newRecur.setEndDate(endDate);
		}
		return newRecur;

	}

	/**
	 * Helper function to create the expected file to compare with the one generated by logic
	 * 
	 * @param expectedFile
	 *            Expected file to save
	 * @throws SAXException
	 * @throws ParseException
	 */
	private void storageCreateExpectedTask(File expectedFile, String description, String date,
			String startTime, String endTime, Boolean completed, Recur recur)
					throws SAXException, ParseException {
		Storage expStorage = new Storage(expectedFile);
		Task newTask = new Task();
		newTask.setDescription(description);

		if (date != null) {
			TaskDate calDate = new TaskDate();
			calDate.setDateFromString(date);
			newTask.setDate(calDate);
		}
		if (startTime != null) {
			TaskTime calStartTime = new TaskTime();
			calStartTime.setTimeFromString(startTime);
			newTask.setStartTime(calStartTime);
		}
		if (endTime != null) {
			TaskTime calEndTime = new TaskTime();
			calEndTime.setTimeFromString(endTime);
			newTask.setEndTime(calEndTime);
		}
		if (completed) {
			newTask.toggleCompleted();
		}
		if (recur != null) {
			newTask.setRecur(recur);
		}
		expStorage.addToTaskList(newTask);
		expStorage.saveTasksToFile();
	}

	/**
	 * Helper function to execute the command through logic and saving it through its storage
	 * 
	 * @param cmd
	 *            command to be parsed and executed
	 * @throws SAXException
	 */
	private void logicExecuteCommand(Logic logic, String cmd) throws SAXException {

		logic.executeCommand(cmd);
		logic.saveTasksToFile();
	}

	@Test
	public final void testAddFloatingTask() throws SAXException, IOException, ParseException {

		// Setting up actual Task List for comparison
		File testFile = new File(TEST_FILE_NAME);
		Logic logic = new Logic(testFile);
		logicExecuteCommand(logic, "CS2103T Reading");

		// Setting up expected Task List for comparison
		File expectedFile = new File(EXPECTED_FILE_NAME);
		storageCreateExpectedTask(expectedFile, "CS2103T Reading", null, null, null, false, null);

		// Settings for XML formatting
		XMLUnit.setIgnoreWhitespace(true);
		XMLUnit.setIgnoreComments(true);
		XMLUnit.setIgnoreDiffBetweenTextAndCDATA(true);
		XMLUnit.setNormalizeWhitespace(true);

		// This is to test the expected behavior of this function
		FileReader fr1 = new FileReader(expectedFile);
		FileReader fr2 = new FileReader(testFile);
		XMLAssert.assertXMLEqual(fr1, fr2);
	}

	@Test
	public final void testAddDeadlineWithNumberInDescription()
			throws SAXException, IOException, ParseException {

		// Setting up actual Task List for comparison
		File testFile = new File(TEST_FILE_NAME);
		Logic logic = new Logic(testFile);
		logicExecuteCommand(logic, "500 words CFG1010 8/4");

		// Setting up expected Task List for comparison
		File expectedFile = new File(EXPECTED_FILE_NAME);
		storageCreateExpectedTask(expectedFile, "500 words CFG1010", "8-4-2016", null, null, false, null);

		// Settings for XML formatting
		XMLUnit.setIgnoreWhitespace(true);
		XMLUnit.setIgnoreComments(true);
		XMLUnit.setIgnoreDiffBetweenTextAndCDATA(true);
		XMLUnit.setNormalizeWhitespace(true);

		// This is to test the expected behavior of this function
		FileReader fr1 = new FileReader(expectedFile);
		FileReader fr2 = new FileReader(testFile);
		XMLAssert.assertXMLEqual(fr1, fr2);
	}

	@Test
	public final void testAddDeadlineWithTime() throws SAXException, IOException, ParseException {

		// Setting up actual Task List for comparison
		File testFile = new File(TEST_FILE_NAME);
		Logic logic = new Logic(testFile);
		logicExecuteCommand(logic, "Plan Jap Trip 30/1/2016 11am");

		// Setting up expected Task List for comparison
		File expectedFile = new File(EXPECTED_FILE_NAME);
		storageCreateExpectedTask(expectedFile, "Plan Jap Trip", "30-1-2016", "11:00AM", null, false, null);

		// Settings for XML formatting
		XMLUnit.setIgnoreWhitespace(true);
		XMLUnit.setIgnoreComments(true);
		XMLUnit.setIgnoreDiffBetweenTextAndCDATA(true);
		XMLUnit.setNormalizeWhitespace(true);

		// This is to test the expected behavior of this function
		FileReader fr1 = new FileReader(expectedFile);
		FileReader fr2 = new FileReader(testFile);
		XMLAssert.assertXMLEqual(fr1, fr2);
	}

	@Test
	public final void testAddEvent() throws SAXException, IOException, ParseException {

		// Setting up actual Task List for comparison
		File testFile = new File(TEST_FILE_NAME);
		Logic logic = new Logic(testFile);
		logicExecuteCommand(logic, "Buy some potatoes 30/1/2016 11am-12pm");

		// Setting up expected Task List for comparison
		File expectedFile = new File(EXPECTED_FILE_NAME);
		storageCreateExpectedTask(expectedFile, "Buy some potatoes", "30-1-2016", "11:00AM", "12:00PM", false,
				null);

		// Settings for XML formatting
		XMLUnit.setIgnoreWhitespace(true);
		XMLUnit.setIgnoreComments(true);
		XMLUnit.setIgnoreDiffBetweenTextAndCDATA(true);
		XMLUnit.setNormalizeWhitespace(true);

		// This is to test the expected behavior of this function
		FileReader fr1 = new FileReader(expectedFile);
		FileReader fr2 = new FileReader(testFile);
		XMLAssert.assertXMLEqual(fr1, fr2);
	}

	@Test
	public final void testAddDeadlineRecurringDay() throws SAXException, IOException, ParseException {

		// Setting up actual Task List for comparison
		File testFile = new File(TEST_FILE_NAME);
		Logic logic = new Logic(testFile);
		logicExecuteCommand(logic, "Go out with girlfriend 1/4 3d 15");

		// Setting up expected Task List for comparison
		File expectedFile = new File(EXPECTED_FILE_NAME);
		storageCreateExpectedTask(expectedFile, "Go out with girlfriend", "01-4-2016", null, null, false,
				createRecur("DAY", 3, "01-4-216", "15-4-2016")); // TODO-change end date to correct behavior

		// Settings for XML formatting
		XMLUnit.setIgnoreWhitespace(true);
		XMLUnit.setIgnoreComments(true);
		XMLUnit.setIgnoreDiffBetweenTextAndCDATA(true);
		XMLUnit.setNormalizeWhitespace(true);

		// This is to test the expected behavior of this function
		FileReader fr1 = new FileReader(expectedFile);
		FileReader fr2 = new FileReader(testFile);
		XMLAssert.assertXMLEqual(fr1, fr2);
	}

	@Test
	public final void testAddDeadlineRecurringWeek() throws SAXException, IOException, ParseException {

		// Setting up actual Task List for comparison
		File testFile = new File(TEST_FILE_NAME);
		Logic logic = new Logic(testFile);
		logicExecuteCommand(logic, "Go out with jully 2/4 1w 2");

		// Setting up expected Task List for comparison
		File expectedFile = new File(EXPECTED_FILE_NAME);
		storageCreateExpectedTask(expectedFile, "Go out with jully", "02-4-2016", null, null, false,
				createRecur("WEEK", 1, "02-4-216", "16-4-2016")); // TODO-change end date to correct behavior

		// Settings for XML formatting
		XMLUnit.setIgnoreWhitespace(true);
		XMLUnit.setIgnoreComments(true);
		XMLUnit.setIgnoreDiffBetweenTextAndCDATA(true);
		XMLUnit.setNormalizeWhitespace(true);

		// This is to test the expected behavior of this function
		FileReader fr1 = new FileReader(expectedFile);
		FileReader fr2 = new FileReader(testFile);
		XMLAssert.assertXMLEqual(fr1, fr2);
	}

	@Test
	public final void testAddDeadlineRecurringEndWithNumber()
			throws SAXException, IOException, ParseException {

		// Setting up actual Task List for comparison
		File testFile = new File(TEST_FILE_NAME);
		Logic logic = new Logic(testFile);
		logicExecuteCommand(logic, "HIMYM 1/1/2027 12:00 1d 50");

		// Setting up expected Task List for comparison
		File expectedFile = new File(EXPECTED_FILE_NAME);
		storageCreateExpectedTask(expectedFile, "HIMYM", "1-1-2027", "12:00PM", null, false,
				createRecur("DAY", 1, "1-1-2027", "20-2-2027")); // TODO-change end date to correct behavior

		// Settings for XML formatting
		XMLUnit.setIgnoreWhitespace(true);
		XMLUnit.setIgnoreComments(true);
		XMLUnit.setIgnoreDiffBetweenTextAndCDATA(true);
		XMLUnit.setNormalizeWhitespace(true);

		// This is to test the expected behavior of this function
		FileReader fr1 = new FileReader(expectedFile);
		FileReader fr2 = new FileReader(testFile);
		XMLAssert.assertXMLEqual(fr1, fr2);
	}

	@Test
	public final void testAddEventRecurringDay() throws SAXException, IOException, ParseException {

		// Setting up actual Task List for comparison
		File testFile = new File(TEST_FILE_NAME);
		Logic logic = new Logic(testFile);
		logicExecuteCommand(logic, "Buy some potatoes 1/5/2016 11am-12pm 1d 10");

		// Setting up expected Task List for comparison
		File expectedFile = new File(EXPECTED_FILE_NAME);
		storageCreateExpectedTask(expectedFile, "Buy some potatoes", "1-5-2016", "11:00AM", "12:00PM", false,
				createRecur("DAY", 1, "1-5-2016", "11-5-2016"));

		// Settings for XML formatting
		XMLUnit.setIgnoreWhitespace(true);
		XMLUnit.setIgnoreComments(true);
		XMLUnit.setIgnoreDiffBetweenTextAndCDATA(true);
		XMLUnit.setNormalizeWhitespace(true);

		// This is to test the expected behavior of this function
		FileReader fr1 = new FileReader(expectedFile);
		FileReader fr2 = new FileReader(testFile);
		XMLAssert.assertXMLEqual(fr1, fr2);
	}

	@Test
	public final void testEditDescription() throws SAXException, IOException, ParseException {

		// Setting up actual Task List for comparison
		File testFile = new File(TEST_FILE_NAME);
		Logic logic = new Logic(testFile);
		logicExecuteCommand(logic, "CS2103T Reading");
		logicExecuteCommand(logic, "e 1 Potato Reading");

		// Setting up expected Task List for comparison
		File expectedFile = new File(EXPECTED_FILE_NAME);
		storageCreateExpectedTask(expectedFile, "Potato Reading", null, null, null, false, null);

		// Settings for XML formatting
		XMLUnit.setIgnoreWhitespace(true);
		XMLUnit.setIgnoreComments(true);
		XMLUnit.setIgnoreDiffBetweenTextAndCDATA(true);
		XMLUnit.setNormalizeWhitespace(true);

		// This is to test the expected behavior of this function
		FileReader fr1 = new FileReader(expectedFile);
		FileReader fr2 = new FileReader(testFile);
		XMLAssert.assertXMLEqual(fr1, fr2);
	}

	@Test
	public final void testEditDate() throws SAXException, IOException, ParseException {

		// Setting up actual Task List for comparison
		File testFile = new File(TEST_FILE_NAME);
		Logic logic = new Logic(testFile);
		logicExecuteCommand(logic, "Tomato Reading 11/4/2016");
		logicExecuteCommand(logic, "e 1 20/4/2016");

		// Setting up expected Task List for comparison
		File expectedFile = new File(EXPECTED_FILE_NAME);
		storageCreateExpectedTask(expectedFile, "Tomato Reading", "20-4-2016", null, null, false, null);

		// Settings for XML formatting
		XMLUnit.setIgnoreWhitespace(true);
		XMLUnit.setIgnoreComments(true);
		XMLUnit.setIgnoreDiffBetweenTextAndCDATA(true);
		XMLUnit.setNormalizeWhitespace(true);

		// This is to test the expected behavior of this function
		FileReader fr1 = new FileReader(expectedFile);
		FileReader fr2 = new FileReader(testFile);
		XMLAssert.assertXMLEqual(fr1, fr2);
	}

	@Test
	public final void testEditStartTime() throws SAXException, IOException, ParseException {

		// Setting up actual Task List for comparison
		File testFile = new File(TEST_FILE_NAME);
		Logic logic = new Logic(testFile);
		logicExecuteCommand(logic, "Tomato Reading 11/4/2016 2pm");
		logicExecuteCommand(logic, "e 1 3pm");

		// Setting up expected Task List for comparison
		File expectedFile = new File(EXPECTED_FILE_NAME);
		storageCreateExpectedTask(expectedFile, "Tomato Reading", "11-4-2016", "3:00PM", null, false, null);

		// Settings for XML formatting
		XMLUnit.setIgnoreWhitespace(true);
		XMLUnit.setIgnoreComments(true);
		XMLUnit.setIgnoreDiffBetweenTextAndCDATA(true);
		XMLUnit.setNormalizeWhitespace(true);

		// This is to test the expected behavior of this function
		FileReader fr1 = new FileReader(expectedFile);
		FileReader fr2 = new FileReader(testFile);
		XMLAssert.assertXMLEqual(fr1, fr2);
	}

	@Test
	public final void testEditStartTimeFromEvent() throws SAXException, IOException, ParseException {

		// Setting up actual Task List for comparison
		File testFile = new File(TEST_FILE_NAME);
		Logic logic = new Logic(testFile);
		logicExecuteCommand(logic, "Cabbage Reading 11/4/2016 2pm-4pm");
		logicExecuteCommand(logic, "e 1 3pm");

		// Setting up expected Task List for comparison
		File expectedFile = new File(EXPECTED_FILE_NAME);
		storageCreateExpectedTask(expectedFile, "Cabbage Reading", "11-4-2016", "3:00PM", null, false, null);

		// Settings for XML formatting
		XMLUnit.setIgnoreWhitespace(true);
		XMLUnit.setIgnoreComments(true);
		XMLUnit.setIgnoreDiffBetweenTextAndCDATA(true);
		XMLUnit.setNormalizeWhitespace(true);

		// This is to test the expected behavior of this function
		FileReader fr1 = new FileReader(expectedFile);
		FileReader fr2 = new FileReader(testFile);
		XMLAssert.assertXMLEqual(fr1, fr2);
	}

	@Test
	public final void testEditEndTime() throws SAXException, IOException, ParseException {

		// Setting up actual Task List for comparison
		File testFile = new File(TEST_FILE_NAME);
		Logic logic = new Logic(testFile);
		logicExecuteCommand(logic, "Cucumber Reading 11/4/2016 2pm-4pm");
		logicExecuteCommand(logic, "e 1 2pm-5pm");

		// Setting up expected Task List for comparison
		File expectedFile = new File(EXPECTED_FILE_NAME);
		storageCreateExpectedTask(expectedFile, "Cucumber Reading", "11-4-2016", "2:00PM", "5:00PM", false,
				null);

		// Settings for XML formatting
		XMLUnit.setIgnoreWhitespace(true);
		XMLUnit.setIgnoreComments(true);
		XMLUnit.setIgnoreDiffBetweenTextAndCDATA(true);
		XMLUnit.setNormalizeWhitespace(true);

		// This is to test the expected behavior of this function
		FileReader fr1 = new FileReader(expectedFile);
		FileReader fr2 = new FileReader(testFile);
		XMLAssert.assertXMLEqual(fr1, fr2);
	}

	@Test
	public final void testEditRecurTimeUnit() throws SAXException, IOException, ParseException {

		// Setting up actual Task List for comparison
		File testFile = new File(TEST_FILE_NAME);
		Logic logic = new Logic(testFile);
		logicExecuteCommand(logic, "Buy some potatoes 1/5/2016 11am-12pm 1d 1");
		logicExecuteCommand(logic, "e 1 1w 1");

		// Setting up expected Task List for comparison
		File expectedFile = new File(EXPECTED_FILE_NAME);
		storageCreateExpectedTask(expectedFile, "Buy some potatoes", "1-5-2016", "11:00AM", "12:00PM", false,
				createRecur("WEEK", 1, "1-5-2016", "8-5-2016"));

		// Settings for XML formatting
		XMLUnit.setIgnoreWhitespace(true);
		XMLUnit.setIgnoreComments(true);
		XMLUnit.setIgnoreDiffBetweenTextAndCDATA(true);
		XMLUnit.setNormalizeWhitespace(true);

		// This is to test the expected behavior of this function
		FileReader fr1 = new FileReader(expectedFile);
		FileReader fr2 = new FileReader(testFile);
		XMLAssert.assertXMLEqual(fr1, fr2);
	}

	@Test
	public final void testEditRecurFrequency() throws SAXException, IOException, ParseException {

		// Setting up actual Task List for comparison
		File testFile = new File(TEST_FILE_NAME);
		Logic logic = new Logic(testFile);
		logicExecuteCommand(logic, "Buy some potatoes 1/5/2016 11am-12pm 1d 1");
		logicExecuteCommand(logic, "e 1 3d 1");

		// Setting up expected Task List for comparison
		File expectedFile = new File(EXPECTED_FILE_NAME);
		storageCreateExpectedTask(expectedFile, "Buy some potatoes", "1-5-2016", "11:00AM", "12:00PM", false,
				createRecur("DAY", 3, "1-5-2016", "4-5-2016"));

		// Settings for XML formatting
		XMLUnit.setIgnoreWhitespace(true);
		XMLUnit.setIgnoreComments(true);
		XMLUnit.setIgnoreDiffBetweenTextAndCDATA(true);
		XMLUnit.setNormalizeWhitespace(true);

		// This is to test the expected behavior of this function
		FileReader fr1 = new FileReader(expectedFile);
		FileReader fr2 = new FileReader(testFile);
		XMLAssert.assertXMLEqual(fr1, fr2);
	}

	@Test
	public final void testEditRecurTimesToRecur() throws SAXException, IOException, ParseException {

		// Setting up actual Task List for comparison
		File testFile = new File(TEST_FILE_NAME);
		Logic logic = new Logic(testFile);
		logicExecuteCommand(logic, "Buy some potatoes 1/5/2016 11am-12pm 1d 1");
		logicExecuteCommand(logic, "e 1 1d 5");

		// Setting up expected Task List for comparison
		File expectedFile = new File(EXPECTED_FILE_NAME);
		storageCreateExpectedTask(expectedFile, "Buy some potatoes", "1-5-2016", "11:00AM", "12:00PM", false,
				createRecur("DAY", 1, "1-5-2016", "6-5-2016"));

		// Settings for XML formatting
		XMLUnit.setIgnoreWhitespace(true);
		XMLUnit.setIgnoreComments(true);
		XMLUnit.setIgnoreDiffBetweenTextAndCDATA(true);
		XMLUnit.setNormalizeWhitespace(true);

		// This is to test the expected behavior of this function
		FileReader fr1 = new FileReader(expectedFile);
		FileReader fr2 = new FileReader(testFile);
		XMLAssert.assertXMLEqual(fr1, fr2);
	}
	
	@Test
	public final void testEditRecurEndOfRecur() throws SAXException, IOException, ParseException {

		// Setting up actual Task List for comparison
		File testFile = new File(TEST_FILE_NAME);
		Logic logic = new Logic(testFile);
		logicExecuteCommand(logic, "Buy some potatoes 1/5/2016 11am-12pm 1d 1/1/2016");
		logicExecuteCommand(logic, "e 1 1d 10/1/2016");

		// Setting up expected Task List for comparison
		File expectedFile = new File(EXPECTED_FILE_NAME);
		storageCreateExpectedTask(expectedFile, "Buy some potatoes", "1-5-2016", "11:00AM", "12:00PM", false,
				createRecur("DAY", 1, "1-5-2016", "10-1-2016"));

		// Settings for XML formatting
		XMLUnit.setIgnoreWhitespace(true);
		XMLUnit.setIgnoreComments(true);
		XMLUnit.setIgnoreDiffBetweenTextAndCDATA(true);
		XMLUnit.setNormalizeWhitespace(true);

		// This is to test the expected behavior of this function
		FileReader fr1 = new FileReader(expectedFile);
		FileReader fr2 = new FileReader(testFile);
		XMLAssert.assertXMLEqual(fr1, fr2);
	}
}
