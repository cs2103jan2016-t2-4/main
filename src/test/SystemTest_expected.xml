<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<wuriTasks>
   <Task>
      <Description>Go out with girlfriend</Description>
      <Date>01/09/16</Date>
      <StartTime/>
      <EndTime/>
      <Completed>no</Completed>
      <recur>
         <timeUnit>DAY</timeUnit>
         <frequency>3</frequency>
         <startOfRecurr>01/09/16</startOfRecurr>
         <endOfRecurr>16/10/16</endOfRecurr>
      </recur>
   </Task>
</wuriTasks>
